draft-ietf-quic-transport.md

```diff
@@ -3440,10 +3466,30 @@ ih:
 : Only a CONNECTION_CLOSE frame of type 0x1c can appear in Initial or Handshake
   packets.
 
-Section 4 of {{QUIC-TLS}} provides more detail about these restrictions.  Note
-that all frames can appear in 1-RTT packets.  An endpoint MUST treat receipt of
-a frame in a packet type that is not permitted as a connection error of type
-PROTOCOL_VIOLATION.
+The rules below generalize those of TLS, in that frames associated with
+establishing the connection can usually appear in packets in any packet number
+space, whereas those associated with transferring data can only appear in the
+application data packet number space:
+
+- PADDING, PING, and CRYPTO frames MAY appear in any packet number space.
+
+- CONNECTION_CLOSE frames signaling errors at the QUIC layer (type 0x1c) MAY
+  appear in any packet number space. CONNECTION_CLOSE frames signaling
+  application errors (type 0x1d) MUST only appear in the application data packet
+  number space.
+
+- ACK frames MAY appear in any packet number space, but can only acknowledge
+  packets that appeared in that packet number space.  However, as noted below,
+  0-RTT packets cannot contain ACK frames.
+
+- All other frame types MUST only be sent in the application data packet number
+  space.
+
+Note that all frames can appear in 1-RTT packets.  Note that it is not possible
+to send the following frames in 0-RTT packets for various reasons: ACK, CRYPTO,
+HANDSHAKE_DONE, NEW_TOKEN, PATH_RESPONSE, and RETIRE_CONNECTION_ID. An endpoint
+MUST treat receipt of a frame in a packet type that is not permitted as a
+connection error of type PROTOCOL_VIOLATION.
 
```

Comment:
This will need to be reconciled with PR #4136, which does not copy the text at this location and updates references. It might be more expedient to not do this change and not modify Section 4 of TLS is this Pr (4138).

Reply:
I'll resolve merge conflicts as they arise, if that PR lands first I'll look at the exact text that was merged and make sure this PR is updated to reflect that